### PR TITLE
Mark start/end of handwritten resources in provider.go

### PR DESCRIPTION
**Pull request by c2thorn**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 1]
```diff
@@ -381,6 +381,7 @@ products.each do |product|
 end     # products.each do
 -%>
 			},
+			// ####### START handwritten resources ###########
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/provider.go.erb:384]: drive-by comment: Want to delineate all 3 sections with the same formatting? That'd make it simple to code a script to count the resources / engine at a given SHA if that was desirable

Comment (c2thorn) [reply mmv1/third_party/terraform/utils/provider.go.erb:385]: good idea, implemented

File: mmv1/third_party/terraform/utils/provider.go.erb [hunk 2]
```diff
@@ -206,6 +206,7 @@ func Provider() *schema.Provider {
 			},
 		},
 
+		// ####### START datasources ###########
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/provider.go.erb:209]: It'd likely be easier for a tool to parse if the start markers were down a line & end markers up a line, such that the only lines between them are the resource lines. It's pretty marginal since they can just skip a line on either side though 🤷 

i.e.

```
// ####### START datasources ###########
"google_active_folder": dataSourceGoogleActiveFolder(),
...
"google_redis_instance": dataSourceGoogleRedisInstance(),
// ####### END datasources ###########
```

**Review by rileykarson — APPROVED** [review]
